classes/submission/Sanitizer.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/**
+ * @file classes/submission/Sanitizer.php
+ *
+ * Copyright (c) 2014-2020 Simon Fraser University
+ * Copyright (c) 2000-2020 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Sanitizer
+ *
+ * @brief A sanitization class to sanitize submission data before saving
+ */
+
+ namespace PKP\submission;
+
+use Exception;
+use Illuminate\Support\Arr;
+use PKP\core\PKPString;
+
+class Sanitizer 
```

Comment:
Right now we only have one input method, through the API, so I'd prefer to see sanitization happen there. For example, we might want to permit a title added through CLI to be less strict in the sanitization. Same with an import. So, in general, I'd like to see sanitization tied to the request handler in some way.

However, if we want a generic sanitization approach for entities separate from the handlers, then I'd suggest merging your approach to sanitization with the existing code that coerces types according to the schema and does other sanitization. You can see this code in `APIHandler::convertStringsToSchema()`. But some consideration will be needed to properly handle data from the API vs CLI vs other input sources.